src/core/dbus-cgroup.c

```diff
@@ -360,33 +360,24 @@ static int bus_cgroup_set_transient_property(
                 return 1;
 
         } else if (streq(name, "DelegateControllers")) {
+                _cleanup_strv_free_ char **l = NULL;
+                char **p;
                 CGroupMask mask = 0;
 
-                r = sd_bus_message_enter_container(message, 'a', "s");
+                r = sd_bus_message_read_strv(message, &l);
```

Comment:
why this change? you don't actually need the list as char**, do you?

I mean, if all you want to check is whether an empty list is assigned, then mask == 0 is a good check, as we add a bit for every item we decode...

Reply:
Indeed. Updated.